common/txmgr/types/models.go

```diff
@@ -0,0 +1,16 @@
+package types
+
+type TxmErrorType int
+
+// Generalized transaction manager error types that dictates what should be the next action, depending on the RPC error response.
+const (
```

Comment:
Since these are just standard error codes returned by the Client Interface, no need to move them to this package.
Keep them inside the client.go file itself.

Reply:
If I move this enum to the client, the Fatal name will conflict with an existing one on the error file (https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/client/errors.go#L57). Any suggestions? How about `Unrecoverable`?

Reply:
@prashantkumar1982 I don't understand your comment - won't these errors be used in other chain-clients as well?

If they are, it wouldn't make sense for them to redefine the error types especially since these are enums or for the solana client to import the evm client to have access to error types

Reply:
Yeah, initially that was my thought as well, that's why I've implemented them as general errors. @aalu1418 do you think the categories we've created here could be used for non-EVM chains as well, i.e. Solana? I do have some concerns about the types and how many we had to create only for the EVM chains.

Reply:
ah here's prashant's reply: https://github.com/smartcontractkit/chainlink/pull/8681#pullrequestreview-1349300151

Reply:
Yes, common client interface, wherever it ends up being created.
I didn't realize it earlier, but looks like Sergei is going to create that in his changes. So Dimitris can just put those there.
I'm ok to leave these anywhere till that is ready.